src/index.ml

```diff
@@ -364,42 +424,51 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
               l "[%s] new entries detected, reading log from disk"
                 (Filename.basename t.root));
           iter_io add_log_entry log.io ~min:log_offset )
-        else if log_offset > new_log_offset then assert false
+        else if log_offset > new_log_offset then
+          (* In that case the log has probably been emptied and is being
+             refilled with async_log contents. *)
+          no_changes ()
```

Comment:
Seems odd to have two `else` cases that both do `no_changes ()`: is this just so that the comment can be here?

Reply:
Yes I think it's worth having a specific case here to highlight some action might be needed (or a comment saying nothing is needded).